lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -1423,6 +1405,36 @@ def write_store(
         )
         return rval
 
+    @router.delete(
+        "/api/workflows/{workflow_id}",
+        summary="Add the deleted flag to a workflow.",
+    )
+    def delete_workflow(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        workflow_id: DecodedDatabaseIdField = StoredWorkflowIDPathParam,
+    ):
+        workflow_to_delete = self.workflows_manager.get_stored_workflow(trans, workflow_id)
+        if workflow_to_delete.user != trans.user and not trans.user_is_admin:
+            raise exceptions.InsufficientPermissionsException()
+        self.workflows_manager.delete(workflow_to_delete)
+        return self.workflow_contents_manager.workflow_to_dict(trans, workflow_to_delete, style="instance")
```

Comment:
I think a 204 is more appropriate here, certainly if you're deleting a bunch of workflows.

Reply:
done in https://github.com/galaxyproject/galaxy/pull/14794/commits/8f0b708698b970e59a023a7190f3b678cee80be2
(although I like returning the actual object more, especially for the undelete call)

Reply:
> done in [8f0b708](https://github.com/galaxyproject/galaxy/commit/8f0b708698b970e59a023a7190f3b678cee80be2) (although I like returning the actual object more, especially for the undelete call)

I'd also prefer to have both calls return the actual deleted/undeleted object with a 200 response status code, which would be more consistent with what we do in many other delete API endpoints.
Agreed that a 204 status code would be fine for a batch delete of a bunch of workflows.

Reply:
But that's bad for performance and I can't think of a reason to look at the return value given that you know exactly what it's going to be. I suspect we inherited this pattern from the mako days where it may have been useful, but we shouldn't keep this pattern. None of the github API endpoints return the whole object if you delete things for instance.